### DISCUSSION TITLE
Looking for wrong table name in custom relation class

### DISCUSSION THREAD

**Discussion by rabol** [laravel/framework#36436] [Q&A]
- Laravel Version: 8.30.0
- PHP Version: 8.0.0
- Database Driver & Version:
mysql  Ver 14.14 Distrib 5.7.32, for Linux (x86_64) using  EditLine wrapper

### Description:
I'm creating a custom model class for a relation.
 when trying to get the related data, and sql error is issued and the select try to select a table that does not exists.

SQLSTATE[42S02]: Base table or view not found: 1146 Table 'xxx.allergy_family_member_plan_setting_family_member_plan_setting' doesn't exist (SQL: select * from `allergy_family_member_plan_setting_family_member_plan_setting` where `family_member_plan_setting_id` = 1)

### Steps To Reproduce:
```
artisan make:model AllergyFamilyMemberPlanSetting -m --pivot
```

this will generate a model and a migration

change the migration:

```
public function up()
    {
        Schema::create('allergy_family_member_plan_setting', function (Blueprint $table) {
            $table->id();
            $table->unsignedBigInteger('allergy_id');
            $table->unsignedBigInteger('family_member_plan_setting_id');
            $table->timestamps();
        });
    }
```

add the relation to the FamilyMemberPlanSetting

```
public function allergies(): BelongsToMany
{
    return $this->belongsToMany(AllergyFamilyMemberPlanSetting::class);
}
```

then try this:

$fmps = FamilyMemberPlanSetting::create()
$fmps->allergies()->sync([]);

the an error like this is issued

```
SQLSTATE[42S02]: Base table or view not found: 1146 Table '<db>.allergy_family_member_plan_setting_family_member_plan_setting' doesn't exist (SQL: select * from `allergy_family_member_plan_setting_family_member_plan_setting` where `family_member_plan_setting_id` = 1)
````

notice the table name is: allergy_family_member_plan_setting_family_member_plan_setting and not 'allergy_family_member_plan_setting'

I have also tried to set the $table property in the model, but that does not make any difference

**Comment by rodrigopedra** [comment]
From your description it seems to me you are after a `->hasMany()` relation instead of a `->belongsToMany()`

The `->belongsToMany()` table expects an auxiliary pivot table, and it is guessing this pivot table name according to the framework's conventions:

https://laravel.com/docs/8.x/eloquent-relationships#many-to-many-table-structure

If I didn’t understood it well and you want to use a `->belongsToMany()`, the method accepts a custom name table as its second parameter when you need a table name different than the framework's convention.

```php
->belongsToMany(Model::class, 'my_custom_table');
```

P.S.: sorry if the formatting is wrong, I am typing from my phone.
